src/MVTSource.js

```diff
@@ -296,10 +299,21 @@ class MVTSource {
    * @param {TileContext} tileContext
    */
   async _fetchTile(tileContext) {
+    // Draw debug tile info before trying to get tile data so that if errors occur, the debug info still draws
+    this._drawDebugInfo(tileContext);
+
+    // Get the tile from the id
     const id = tileContext.parentId || tileContext.id;
     const tile = getTileFromString(id);
 
-    const src = this._url.replace('{z}', tile.zoom).replace('{x}', tile.x).replace('{y}', tile.y);
+    // Create the source url with the urlGenerator function if it exists, else replace {z} {y} and {x}
+    let src;
+    if (this._urlGenerator) {
+      src = this._urlGenerator(tile.zoom, tile.x, tile.y);
```

Comment:
Do we anticipate `this._urlGenerator` ever needing access to `this._url`, like with the fallback strategy?